webapp/src/client.js

```diff
@@ -1,20 +1,25 @@
 import {id as pluginId} from './manifest';
 
+const HEADER_X_CSRF_TOKEN = 'X-CSRF-Token';
+
 export class Client {
     constructor() {
         this.url = `/plugins/${pluginId}/api/v1`;
     }
 
     connected = () => {
-        return this.doFetch(`${this.url}/connected`, {
-            method: 'POST',
-            headers: {
-                'X-Requested-With': 'XMLHttpRequest',
```

Comment:
I think we still can add the `X-Requested-With` header optionally - 0/5

Reply:
I can re-add that. I took it out since I didn't understand the exact purpose of the header before.

It sounds like `X-Requested-With` is sufficient to protect against CSRF attacks though, so why do we print warnings for plugins that only use `X-Requested-With`?

Reply:
I think the OWASP Cheat Sheet explains nicely why it's often enough, but what's the problem with that approach: https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md#use-of-custom-request-headers

If we wouldn't print a warning, developers would never migrate to the actually more secure method.

Reply:
Okay, so the header is usually enough, but it relies on the browser which can have holes occasionally? I've re-added it anyway, but I'd like to properly understand what we're doing here for my own knowledge :)